### PR TITLE
HPCC-21856 Filter popup displayed when not visible

### PR DESCRIPTION
**Pull request by miguelvazq**
Whenever a user clicks on DFUQueryWidget and they exit that widget prematurely the filter popup will still popup. This was also happening when searching through global search

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 1]
```diff
@@ -441,7 +441,14 @@ define([
                 });
 
                 this.initWorkunitsGrid();
-                this.checkIfWarning();
+               
+                !params.searchResults ? this.checkIfWarning() : null;
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: Should be a simple `if`.

**Comment by richardkchapman** [comment]
@miguelvazq This should probably target 7.0.x

**Review by GordonSmith — CHANGES_REQUESTED** [review]
I think the warning should only show once - but this looks like it will show every time user tabs to the page?

**Comment by miguelvazq** [comment]
The warning will show if they come back to the tab and the filter is not set.

**Comment by miguelvazq** [comment]
> @miguelvazq This should probably target 7.0.x



> @miguelvazq This should probably target 7.0.x

@richardkchapman there is a fix that @GordonSmith put in that looks like is in 7.2 which is why I added here.

**Comment by miguelvazq** [comment]
@GordonSmith updated and comment added.

**Comment by GordonSmith** [comment]
That fix originally went into: community_7.0.26-1:
image

**Comment by miguelvazq** [comment]
Changed base to 7.0.x

**Review by GordonSmith — APPROVED** [review]
Looks fine.

**Pull request merged**